File: 范例/Accpt/12XX/p1284.pas
PROGRAM p1284;

VAR
        n       :longint;


FUNCTION check(m:longint):boolean;
var
        i       :longint;
        sum     :longint;
begin
sum:=1;
for i:=2 to trunc(sqrt(m)) do
        if m mod i=0 then
                begin
                sum:=sum+i;
                sum:=sum+(m div i);
                end;
if trunc(sqrt(m))=sqrt(m) then dec(sum,trunc(sqrt(m)));
if sum>m then
        begin
        check:=false;
        writeln(m:5,'  ABUNDANT');
        end else
if sum<m then
        begin
        check:=false;
        writeln(m:5,'  DEFICIENT');
        end else check:=true;

end;

BEGIN
read(n);
writeln('PERFECTION OUTPUT');
while n<>0 do
        begin
        if check(n) then writeln(n:5,'  PERFECT');
        read(n);
        end;
writeln('END OF OUTPUT');
END.

File: Vol.03(1200-1299)/#1284 Perfection/p1284.cpp
#include <cstdio>

int main() {
	int	N;
	printf( "PERFECTION OUTPUT\n" );
	while( scanf( "%d", &N ) != EOF ) {
		if( !N ) break;
		int	sum = 0;
		for( int i = N >> 1; i; i-- )
			if( !( N % i ) ) sum += i;
		printf( "%5d  ", N );
		if( sum == N )
			printf( "PERFECT\n" );
		else if( sum < N )
			printf( "DEFICIENT\n" );
		else	printf( "ABUNDANT\n" );
	}
	printf( "END OF OUTPUT\n" );
}
